processing/src/main/java/org/apache/druid/segment/nested/NestedFieldLiteralColumnIndexSupplier.java

```diff
@@ -108,8 +108,14 @@ public NestedFieldLiteralColumnIndexSupplier(
   public <T> T as(Class<T> clazz)
   {
     if (clazz.equals(NullValueIndex.class)) {
-      // null index is always 0 in the global dictionary, even if there are no null rows in any of the literal columns
-      return (T) (NullValueIndex) () -> new SimpleImmutableBitmapIndex(bitmaps.get(0));
+      final BitmapColumnIndex nullIndex;
+      if (dictionary.get(0) == 0) {
```

Comment:
it would have been nicer to have these constants (RHS) defined at the top - something like GLOBAL_DICTIONARY_NULL_INDEX. Some other day perhaps.